component/service/src/main/java/org/exoplatform/social/rest/impl/identity/IdentityRestResourcesV1.java

```diff
@@ -440,4 +434,56 @@ public Response getRelationshipsOfIdentity(@Context UriInfo uriInfo,
     }
     return EntityBuilder.getResponse(collectionRelationship, uriInfo, RestUtils.getJsonMediaType(), Response.Status.OK);
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("{id}/identities/common")
+  @RolesAllowed("users")
+  @ApiOperation(value = "Gets common connections with identities",
+          httpMethod = "GET",
+          response = Response.class)
+  @ApiResponses(value = {
+          @ApiResponse(code = 200, message = "Request fulfilled"),
+          @ApiResponse(code = 500, message = "Internal server error"),
+          @ApiResponse(code = 400, message = "Invalid query input"),
+          @ApiResponse(code = 412, message = "Precondition failed, check your input params")})
+  public Response getCommonConnections(@Context UriInfo uriInfo,
+                                       @ApiParam(value = "Relationship id", required = true) @PathParam("id") String id,
+                                       @ApiParam(value = "Offset", required = false, defaultValue = "0") @QueryParam("offset") int offset,
+                                       @ApiParam(value = "Limit", required = false, defaultValue = "20") @QueryParam("limit") int limit,
+                                       @ApiParam(value = "Returning the number of common connections or not", defaultValue = "false") @QueryParam("returnSize") boolean returnSize,
+                                       @ApiParam(value = "Asking for a full representation of a specific subresource if any", required = false) @QueryParam("expand") String expand) throws Exception {
+
+    try {
+      RelationshipManager relationshipManager = CommonsUtils.getService(RelationshipManager.class);
+      int commonConnectionsSize = 0;
+
+      Identity authenticatedUser = CommonsUtils.getService(IdentityManager.class).getOrCreateIdentity(OrganizationIdentityProvider.NAME, ConversationState.getCurrent().getIdentity().getUserId());
+
+      List<Identity> currentUserConnections = Arrays.asList(relationshipManager.getConnections(authenticatedUser).load(0, 0));
+
+      Identity relationship = CommonsUtils.getService(IdentityManager.class).getIdentity(id, true);
+      List<Identity> relationshipConnections = Arrays.asList(relationshipManager.getConnections(relationship).load(0, 0));
+
+      List<Identity> list = new ArrayList<>();
```

Comment:
Java 8 streams can be used here to get the commons elements.

```suggestion
      List<Identity> commonConnections = currentUserConnections.stream().filter(withConnections::contains)
                     .collect(Collectors.toList());
```